src/checkTargetMatchToPR.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+const semverDiff = require('semver/functions/diff')
+
+const { targetOptions } = require('./getTargetInput')
+
+const checkTargetMatchToPR = (prTitle, target) => {
+  if (target === targetOptions.major) {
+    return true
+  }
+  const expression = new RegExp(/from ([^\s]+) to (.+)/g)
+  const match = expression.exec(prTitle)
+
+  if (match !== null) {
+    const prTarget = semverDiff(match[1], match[2])
+    if (prTarget) {
+      if (prTarget === targetOptions.major) {
+        return false
+      }
+      if (prTarget === targetOptions.minor && target === targetOptions.patch) {
+        return false
+      }
+    }
+  }
+  return true
```

Comment:
1. prefer returning early. if match === null -> return true
2. from the docs

> diff(v1, v2): Returns difference between two versions by the release type (major, premajor, minor, preminor, patch, prepatch, or prerelease), or null if the versions are the same.

it doesn't look like we're accounting for all the possible release types here